examples/sil/carbon-aware_control_unit/carbon_aware_control_unit.py

```diff
@@ -49,28 +50,50 @@ def __init__(self, server_address: str, nodes: dict) -> None:
         self.power_modes = ["power-saving", "normal", "high performance"]
         self.nodes = nodes
         self.client = HTTPClient(server_address)
-
+        self.nodes_power_mode = {}
         self.battery = RemoteBattery()
         self.ci = 0.0
         self.solar = 0.0
+        self.lock = Lock()
```

Comment:
what do we need this Lock for?

Reply:
The battery soc, solar and ci values are accessed by two seperate threads at the same time. While this is not a direct problem since the values are not changed at the same time, it is possible that in `scenario_step()` some values were updated while the rest was not yet -> With the lock, we can be sure that the values are updated and read together.

Reply:
As I think about it now, The `scenario_step()` is now always waiting for the update function to finish since the values are locked right? Then the lock should indeed go...